test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout.lua

```diff
@@ -0,0 +1,917 @@
+c.getHMIEventName = commonRC.getHMIEventName
+c.getHMIRequestParams = commonRC.getHMIRequestParams
+c.getHMIResponseParams = commonRC.getHMIResponseParams
+c.getConfigAppParams = actions.getConfigAppParams
+c.setSDLIniParameter = actions.setSDLIniParameter
+
+c.notificationTime = 0
+c.jsonFileToTable = utils.jsonFileToTable
+c.tableToJsonFile = utils.tableToJsonFile
+c.cloneTable = utils.cloneTable
+c.Step = runner.Step
+c.Title = runner.Title
+
+c.modules = { "RADIO", "CLIMATE" }
+
+c.rpcs = {}
+
+c.rpcsArray = {
+  "SendLocation",
+  "Alert",
+  "SubtleAlert",
+  "PerformInteraction",
+  "Slider",
+  "Speak",
+  "ScrollableMessage",
+  "DiagnosticMessage",
+  "SetInteriorVehicleData",
+  "CreateInteractionChoiceSet",
+  "DeleteInteractionChoiceSet",
+  "DeleteSubMenu",
+  "AlertManeuver",
+  "AddCommand",
+  "ChangeRegistration",
+  "SetGlobalProperties",
+  "AddSubMenu"
+}
+
+c.rpcsArrayWithCustomTimeout = {
+  ["PerformInteraction"] = { timeout = 5000 },
+  ["ScrollableMessage"] = { timeout = 1000 },
+  ["Alert"] = { timeout = 3000 },
+  ["SubtleAlert"] = { timeout = 3000 },
+  ["Slider"] = { timeout = 1000 }
+}
+
+c.defaultTimeout = actions.sdl.getSDLIniParameter("DefaultTimeout")
+c.defaultTimeoutCompensation = actions.sdl.getSDLIniParameter("DefaultTimeoutCompensation")
+
+--[[ Common Functions ]]
+
+--[[ @updatePreloadedPT: Update preloaded file with additional permissions
+--! @parameters: none
+--! @return: none
+--]]
+local function updatePreloadedPT()
+  local pt = actions.sdl.getPreloadedPT()
+  pt.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null
+  local additionalRPCs = {
+    "SendLocation", "DialNumber", "DiagnosticMessage"
+  }
+  pt.policy_table.functional_groupings.NewTestCaseGroup = { rpcs = { } }
+  for _, v in pairs(additionalRPCs) do
+    pt.policy_table.functional_groupings.NewTestCaseGroup.rpcs[v] = {
+      hmi_levels = { "BACKGROUND", "FULL", "LIMITED" }
+    }
+  end
+
+  --insert applications into "app_policies"
+  for i = 1, 2 do
+    local policyAppId = c.getConfigAppParams(i).fullAppID
+    pt.policy_table.app_policies[policyAppId] = c.cloneTable(pt.policy_table.app_policies.default)
+    pt.policy_table.app_policies[policyAppId].groups = { "Base-4", "NewTestCaseGroup", "Navigation-1" }
+    pt.policy_table.app_policies[policyAppId].moduleType = c.modules
+    if c.getConfigAppParams(i).appHMIType[1] == "REMOTE_CONTROL" then
+      pt.policy_table.app_policies[policyAppId].moduleType = c.modules
+      pt.policy_table.app_policies[policyAppId].AppHMIType = { "REMOTE_CONTROL" }
+      table.insert(pt.policy_table.app_policies[policyAppId].groups, "RemoteControl")
+    end
+  end
+
+  actions.sdl.setPreloadedPT(pt)
+end
+
+--[[ @preconditions: Remove policy DB, Logs, create backup and update for preloaded file
+--! @parameters: none
+--! @return: none
+--]]
+function c.preconditions()
+  actions.preconditions()
+  actions.sdl.setSDLIniParameter("DefaultTimeoutCompensation", "0")
```

Comment:
I think instead of adding this line the additional padding should be removed from the OnResetTimeout notifications in all the applicable tests

Reply:
@ShobhitAd Please notice the change described in https://github.com/smartdevicelink/sdl_core/pull/3726#issuecomment-888553791 was suggested and implemented after the scripts has been reviewed. 
This update in Core requires quite a lot of updates in existing scripts for the feature since the expected timeouts are hard-coded. And also additional verification. In order to avoid these updates and to avoid possible mistakes we decided to set `DefaultTimeoutCompensation` to zero assuming all existing behavior should not be changed and SDL should work as before. Additionally we developed 2 new scripts to cover the logic with non-zero `DefaultTimeoutCompensation`.